corehq/apps/app_manager/models.py

```diff
@@ -2338,6 +2338,15 @@ def validate_for_build(self):
                 'type': 'no forms or case list',
                 'module': self.get_module_info(),
             })
+
+        try:
+            get_select_chain(self.get_app(), self)
+        except SuiteValidationError:
+            errors.append({
+                'type': 'circular case hierarchy',
+                'module': self.get_module_info(),
+            })
+
```

Comment:
if it's not too hard it would be great to actually prevent someone from saving an update to a module when a circular reference is detected